pkgs/top-level/all-packages.nix

```diff
@@ -6762,7 +6762,11 @@ with pkgs;
 
   ocropus = callPackage ../applications/misc/ocropus { };
 
-  inherit (callPackages ../development/interpreters/perl {}) perl perl522 perl524 perl526;
+  inherit (
+    if stdenv.buildPlatform == stdenv.hostPlatform
+    then (callPackages ../development/interpreters/perl {})
+    else (callPackages ../development/interpreters/perl-cross {})
+  ) perl perl522 perl524 perl526;
```

Comment:
Is it too bold to make the perl-cross build system overlay the default build environment for native too? (In any case, it can wait, I'm just curious.)

Reply:
I like the idea of merging them, and just pushed fixes that get this working by (among other minor fixes) grabbing most of the existing expression.

I don't think using it by default will go over well-- I thought this came up before but can't find the discussion now...

Of course it would be convenient if we could use the same for both!